src/execution/operator/csv_scanner/buffer_manager/csv_file_handle.cpp

```diff
@@ -31,9 +32,19 @@ bool CSVFileHandle::CanSeek() {
 	return can_seek;
 }
 
-void CSVFileHandle::Seek(idx_t position) {
+void CSVFileHandle::Seek(void *buffer, idx_t nr_bytes, idx_t position) {
 	if (!can_seek) {
-		throw InternalException("Cannot seek in this file");
+		if (is_pipe) {
+			throw InternalException("Can't reconstruct the buffer from a on disk file.");
+		}
+		// If we can't seek in this file, we reset it and re-read up to the necessary point.
+		// This should only happen on extreme cases of memory pressure
+		file_handle->Reset();
+		D_ASSERT(position % nr_bytes == 0);
+		for (idx_t i = 0; i < position / nr_bytes; i++) {
```

Comment:
Can this be turned into an `InternalException` - this should not happen with the new buffer counting mechanism right?